### PR TITLE
Fix application of styles in `Cesium3DTileset`

### LINKED ISSUE #10563: `Cesium3DTileset` won't apply style if `show = false`
Found this while writing unit tests for #10560. `Cesium3DTilesTester.expectRender` and `expectRenderBlank` set `tileset.show = false` first to confirm  the tileset won't render, then does `tileset.show = true` to proceed with the test. The problem is, if`tileset.style` is set while `tileset.show` is false, the style will never actually get applied to the tileset's content. (I put a breakpoint in the `applyStyle` function in `Cesium3DTileStyleEngine`, never got triggered.) Then the style's dirty variable gets reset (`Cesium3DTileStyleEngine.resetDirty`), so it won't be dirty by the time that `tileset.show` is set to true.

### PR DESCRIPTION
**Pull request by j9liu**
This fixes #10563. [Sandcastle for testing.](http://localhost:8080/Apps/Sandcastle/index.html#c=dVJNb9swDP0rgneoDQRyhhXFkLrBtm6HAi1aLMZ28UWR6FQYLQaS7PQD/e+l7DRot04XWXzvkY+kNbkQxWBhB16cCQc7cQ7B9p38NcbyJtPj+5xcVNaBb7KZeGycEBG858iNp8Ea8IsXofagIvwmj6aeKHkxa9xTcdq4xumxYLQIAeLbitP16Xs9gflYpSzBqTXCFRnAH3db8LYDtoILEX0Ps8TpPR6qX5D7CYF6r0G2nrqvgVNdmPx4/vnkpBjp4ZZ2C9EqDHCwtTckQ7xH+L+tVYInY5qQuGmeT/rIjz607ZzP8fyoaLJD3mmyMmhwILfs3UY7QJDKmHxfMxH3tAeirqbXQONWyhmtQkRIopoI18p/62Mkx8upabNhw6PttJm2dzpahor9kl764p65rTSy09FcNsuqUbVMtHS+2G5LPqZp5lKWEbot8iJDue71H06hQ0iOErUqX0srYwdhzdk7v4rQqEJgpO0RV/aBPS6rkvn/SJGUsW5zPYBHdZ9otx+Xl1NQSlmV/HxfGaeR/JX5GQ)

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @j9liu!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/tileset-style-fix/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: Source/Scene/Cesium3DTileset.js [hunk 1]
```diff
@@ -2340,7 +2341,13 @@ Cesium3DTileset.prototype.postPassesUpdate = function (frameState) {
   cancelOutOfViewRequests(this, frameState);
   raiseLoadProgressEvent(this, frameState);
   this._cache.unloadTiles(this, unloadTile);
-  this._styleEngine.resetDirty();
+
+  // If the style wasn't able to be applied this frame, keep it dirty so
+  // the engine can try to apply the style next frame.
```
Comment (ptrgags) [inline Source/Scene/Cesium3DTileset.js:2346]: I would mention that a hidden tileset is one example of when the style wouldn't be applied.

**Review by ptrgags — COMMENTED** [review]
@j9liu looks good, I had one suggestion about documentation before I merge

**Comment by ptrgags** [comment]
Looks good, I'll merge. thanks @j9liu!

**Comment by sanjeetsuhag** [comment]
@ptrgags The prettier check is failing in the branch causing CI to fail in main.

**Comment by ptrgags** [comment]
@sanjeetsuhag ah okay. Since @jjhembd's PR https://github.com/CesiumGS/cesium/pull/10520 is just about to be merged, how about we fix it there since it's just a minor whitespace change?

**Comment by sanjeetsuhag** [comment]
@ptrgags That's fine. Just want to address this since it should never happen with our `pre-commit` hook. Seems like `.md` files are ignored in that. We need to fix that.

**Pull request merged**